virtual_rainforest/models/abiotic/abiotic_tools.py

```diff
@@ -0,0 +1,96 @@
+"""The ``models.abiotic.abiotic_tools`` module contains a set of general functions that
+are shared across submodules in the :mod:`~virtual_rainforest.models.abiotic` model.
+
+TODO cross-check with pyrealm for duplication/ different implementation
+"""  # noqa: D205, D415
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+def calculate_molar_density_air(
+    temperature: NDArray[np.float32],
+    atmospheric_pressure: NDArray[np.float32],
+    standard_mole: float,
+    standard_pressure: float,
+    celsius_to_kelvin: float,
+) -> NDArray[np.float32]:
+    """Calculate temperature-dependent molar density of air.
+
+    Implementation after :cite:t:`maclean_microclimc_2021`.
+
+    Args:
+        temperature: Air temperature, [C]
+        atmospheric_pressure: Atmospheric pressure, [kPa]
+        standard_mole: Moles of ideal gas in 1 m^3 air at standard atmosphere
+        standard_pressure: Standard atmospheric pressure, [kPa]
+        celsius_to_kelvin: Factor to convert temperature in Celsius to absolute
+            temperature in Kelvin
+
+    Returns:
+        molar density of air, [mol m-3]
+    """
+
+    temperature_kelvin = temperature + celsius_to_kelvin
```

Comment:
I don't think you need to pass in `celsius_to_kelvin` as an argument as it'll never change.

Scipy has a function for doing this directly: `from scipy.constants.constants import C2K`

Reply:
The same *might* apply for some of these other constants.

Reply:
I agree, this makes sense.

Reply:
I moved this to the `CoreConsts` now, how else could I call it or where would I import scipy?

Reply:
I guess what I meant was that you probably didn't need to make the physical constants arguments. It just makes it a bit harder to read and `standard_pressure` etc. aren't going to change (right?).

That said, if it makes things easier to test, then it's fine.